### PR TITLE
Fix unit conversion of limits from Fnu to Flam

### PR DESCRIPTION
**Pull request by astrofrog**
### Description

I have expanded the test suite for ``to_unit`` and found that an error occurred when converting limits from Fnu to Flam. The issue in this specific case was that ``if '_pixel_scale_factor' in spec.meta:`` was  ``True`` so the clause relating to ``len(values)==2`` was never invoked, resulting in an issue with incompatible dimensions.

In addition to fixing this logic, I have made it so that we just use one wavelength/frequency in this case for the conversion, as it doesn't specifically make sense to use the first and last spectral axis value. When converting limits we are probably interested in the values being right at the start of the spectral axis.

This builds on top of https://github.com/spacetelescope/jdaviz/pull/2888 and should be rebased once https://github.com/spacetelescope/jdaviz/pull/2888 is merged.

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/app.py [hunk 1]
```diff
@@ -109,47 +109,40 @@ def to_unit(self, data, cid, values, original_units, target_units):
             except RuntimeError:
                 eqv = []
             else:
+
+                # If there are only two values, this is likely the limits being converted, so then
+                # in case we need to use the spectral density equivalency, we need to provide only
+                # to spectral axis values. If there is only one value
+                if not np.isscalar(values) and len(values) == 2:
```
Comment (pllim) [inline jdaviz/app.py:116]: Is this safer?
```suggestion
                if not np.isscalar(values) and len(values) > 1:
```

File: jdaviz/tests/test_app.py [hunk 2]
```diff
@@ -224,13 +224,51 @@ def test_to_unit(cubeviz_helper):
...
+
+    values = [1, 2]
+    original_units = u.MJy
+    target_units = u.erg / u.cm**2 / u.s / u.AA
+
+    new_values = uc.to_unit(cubeviz_helper, data, cid, values, original_units, target_units)
+
+    # In this case we do a regular spectral density conversion, but using the
+    # first value in the spectral axis for the equivalency
+    assert np.allclose(new_values,
```
Comment (pllim) [inline jdaviz/tests/test_app.py:271]: Why not use `astropy.tests.helper.assert_quantity_allclose` here?

File: jdaviz/app.py (hunk 1 above)
Comment (astrofrog) [reply jdaviz/app.py:116]: No it's only for length 2 arrays, which is when we convert min/max ranges

File: jdaviz/tests/test_app.py (hunk 2 above)
Comment (astrofrog) [reply jdaviz/tests/test_app.py:271]: new_values is a plain Numpy array

File: jdaviz/app.py (hunk 1 above)
Comment (astrofrog) [reply jdaviz/app.py:116]: It's a little more complicated - in fact the code as written, without this clause, will only work properly if an array with the same length as the spectral axis is passed in. This clause makes it possible to also deal with the case where min/max values are passed in in which case we use the first spectral axis value in the conversion. But if we were to pass a length 5 array and the spectra axis had length 10 it's not clear what should happen.

In any case I think this discussion belongs in a separate issue/PR as it's beyond the scope of the simple fix here which preserves the original length 2 check.

**Review by gibsongreen — APPROVED** [review]
Post-rebase of pixel scale factor updated PR: #2860:
- New limits test passing
- small medication to conditional to include update from this PR: https://github.com/spacetelescope/jdaviz/blob/c26c5e15518b07248e40bbee338d9729df52d0bc/jdaviz/app.py#L148
- Profile-viewer behavior post display units/conversion changes functioning

**Comment by gibsongreen** [comment]
Note: 
- 3.12 Test failures (11 total) same as #2888

**Pull request merged**